admin_pages/messages/Messages_Admin_Page.core.php

```diff
@@ -4325,7 +4325,13 @@ protected function _setup_response_message_for_deactivating_messenger_with_messa
         }
 
         // if messenger was html or message type was invoice then let's make sure we deactivate invoice payment method.
-        if ($messenger->name === 'html' || $message_type_name === 'invoice') {
+        if (
+            $messenger->name === 'html'
+            && (
+                is_null($message_type)
```

Comment:
hmmmm...
can't fathom why this would fix the issue...
that said the messages system is a bit tricky to grok, so... 🤷🏻 
I want to have a bit more of a look into this, so just going to hold off on this for now.

Reply:
First part of the if condition was always true if you disable one of the HTML message types. It's why the Invoice payment method got disabled after disabling receipt and ticket message types.

I changed the condition to disable the Invoice only when HTML messenger got disabled ( `$messenger->name === 'html' && is_null($message_type)` ) OR invoice message type got disabled ( `$messenger->name === 'html' && $message_type === 'invoice'` ).

It fixed those issues and also based on my E2E tests, everything works fine after disabling the receipt or ticket message types.

Reply:
ug... i read the previous code wrong when I first looked at this

ya sry, this is fine, thnx for the explanation 👍🏻